### ISSUE TITLE
Simplify Component by moving Vue functionality to child VueComponent

### ISSUE THREAD

**Issue by MarcSkovMadsen** [paulopes/panel-components#12]
The `Component` class contains functionality which is only relevant for users of Vue. That can be overwhelming, confusing and/ or hard to maintain. It also makes it difficult to split documentation and examples into something that gradually builds into an understanding (I claim 😄 ).

I would suggest to simplify things by implementing a `VueComponent` child class that can be used by Vue users. 

I believe that would include moving the `data_prefix`, `data_postfix`, `component_data` and related functionality from the `Component` into a `VueComponent`.

**Comment by paulopes** [comment]
Good point. I had thought that a `panel-vue` package would have been too small, but you are right, separating it would be cleaner.

Another thing to consider is that with Vue version 3 this `vue.py` module will probably not work anymore.

**Comment by MarcSkovMadsen** [comment]
My idea would still be to have a VueComponent and ReactComponent in the panel-componentS package.
